src/sqe.rs

```diff
@@ -347,6 +383,33 @@ impl<'a> SubmissionQueueEvent<'a> {
 unsafe impl<'a> Send for SubmissionQueueEvent<'a> { }
 unsafe impl<'a> Sync for SubmissionQueueEvent<'a> { }
 
+pub struct AcceptParams {
+    storage: mem::MaybeUninit<nix::sys::socket::sockaddr_storage>,
+    len: usize,
+}
+
+impl AcceptParams {
+    pub fn uninit() -> Self {
+        let storage = mem::MaybeUninit::uninit();
+        let len = mem::size_of::<nix::sys::socket::sockaddr_storage>();
+        AcceptParams {
+            storage,
+            len
+        }
+    }
+
+    pub unsafe fn as_socket_addr(&self) -> io::Result<SockAddr> {
```

Comment:
I want to make sure I have a correct understanding of this API: this function is safe to call after as the `prep_accept` SQE you passed this to has completed, and calling it before it has completed would be incorrect. Is that right?

Reply:
Absolutely right.